include/swift/AST/DiagnosticsSema.def

```diff
@@ -2995,33 +2995,33 @@ ERROR(overriding_decl_missing_differentiable_attr,none,
 NOTE(protocol_witness_missing_differentiable_attr,none,
      "candidate is missing attribute '%0'", (StringRef))
 
-// @differentiating
-ERROR(differentiating_attr_expected_result_tuple,none,
-      "'@differentiating' attribute requires function to return a two-element tuple of type "
+// @derivative
+ERROR(derivative_attr_expected_result_tuple,none,
+      "'@derivative(of:)' attribute requires function to return a two-element tuple of type "
```

Comment:
One topic to discuss is whether to show `@derivative(of:)` or `@derivative` in user-facing messages.
I chose `@derivative(of:)`, following [the precedent of `dynamicReplacement(for:)`](https://github.com/apple/swift/blob/d2e1f0916976583e6bba20cf550b9d9cfdb62612/include/swift/AST/DiagnosticsSema.def#L4219).

Code comments mostly use `@derivative` without the label.

Reply:
`@dynamicReplacement(for:)` doesn't take additional arguments like we do, but `@derivative(or:wrt:)` is not good either. This works for me.